go-controller/pkg/cni/ovs_unit_test.go

```diff
@@ -0,0 +1,406 @@
+		},
+	}
+
+	for i, tc := range tests {
+		t.Run(fmt.Sprintf("%d:%s", i, tc.desc), func(t *testing.T) {
+			call := mockKexecIface.On(tc.onRetArgs.OnCallMethodName)
+			for _, arg := range tc.onRetArgs.OnCallMethodArgType {
+				call.Arguments = append(call.Arguments, mock.AnythingOfType(arg))
+			}
+			for _, elem := range tc.onRetArgs.RetArgList {
+				call.ReturnArguments = append(call.ReturnArguments, elem)
+			}
+			call.Once()
+
+			e := setExec(mockKexecIface)
+			assert.Equal(t, e, tc.expectedErr)
+			mockKexecIface.AssertExpectations(t)
+		})
+	}
+}
+
+func TestOvsExec(t *testing.T) {
+	mockKexecIface := new(mock_k8s_io_utils_exec.Interface)
+	mockCmd := new(mock_k8s_io_utils_exec.Cmd)
+
+	tests := []struct {
+		desc                string
+		expectedErr         error
+		onRetArgsKexecIface *ovntest.TestifyMockHelper
+		onRetArgsCmdList    *ovntest.TestifyMockHelper
+		runnerInstance      kexec.Interface
+	}{
+		{
+			desc:           "Test codepath when runner is nil and returns error",
+			expectedErr:    nil,
+			runnerInstance: nil,
+		},
+		{
+			desc:                "Test codepath when runner is non nil and returns error",
+			expectedErr:         nil,
+			onRetArgsKexecIface: &ovntest.TestifyMockHelper{"Command", []string{"string", "string"}, []interface{}{mockCmd}},
+			onRetArgsCmdList:    &ovntest.TestifyMockHelper{"CombinedOutput", []string{}, []interface{}{nil, fmt.Errorf("failed to run 'ovs-vsctl")}},
+			runnerInstance:      mockKexecIface,
+		},
+		{
+			desc:                "Test codepath when runner is not nil and does not return an error",
+			expectedErr:         nil,
+			onRetArgsKexecIface: &ovntest.TestifyMockHelper{"Command", []string{"string", "string"}, []interface{}{mockCmd}},
+			onRetArgsCmdList:    &ovntest.TestifyMockHelper{"CombinedOutput", []string{}, []interface{}{nil, nil}},
+			runnerInstance:      mockKexecIface,
+		},
+	}
+	for i, tc := range tests {
+		t.Run(fmt.Sprintf("%d:%s", i, tc.desc), func(t *testing.T) {
+			if tc.onRetArgsKexecIface != nil {
+				ifaceCall := mockKexecIface.On(tc.onRetArgsKexecIface.OnCallMethodName)
+				for _, arg := range tc.onRetArgsKexecIface.OnCallMethodArgType {
+					ifaceCall.Arguments = append(ifaceCall.Arguments, mock.AnythingOfType(arg))
+				}
+				for _, ret := range tc.onRetArgsKexecIface.RetArgList {
+					ifaceCall.ReturnArguments = append(ifaceCall.ReturnArguments, ret)
+				}
+				ifaceCall.Once()
+			}
+			if tc.onRetArgsCmdList != nil {
+				mockCall := mockCmd.On(tc.onRetArgsCmdList.OnCallMethodName)
+				for _, arg := range tc.onRetArgsCmdList.OnCallMethodArgType {
+					mockCall.Arguments = append(mockCall.Arguments, mock.AnythingOfType(arg))
+				}
+				for _, ret := range tc.onRetArgsCmdList.RetArgList {
+					mockCall.ReturnArguments = append(mockCall.ReturnArguments, ret)
+				}
+				mockCall.Once()
+			}
+
+			runner = tc.runnerInstance
+
+			_, e := ovsExec()
+
+			if tc.expectedErr != nil {
+				assert.Error(t, e)
+			}
```

Comment:
maybe better to add else statement as well to ensure that the error returned would be `nil`
`assert.Nil(t, e)`